### PR TITLE
Update google_tag_manager.eno

### PR DESCRIPTION
**Pull request by BrianClifton**
Fine tune of filters as currently it is too blunt, wrongly classifying marketing pixels from Google Tags as "essential" from GTM.

e.g. https://www.googletagmanager.com/a?id=AW-10952806841&v=3..... is a Google Tag for Ads.	

Same test site used as for PR #406.

Note: Regex would be better here as the id param may not always be the first. However that appears to be an edge case so far.

### CODE REVIEW

File: db/patterns/google_tag_manager.eno [hunk 1]
```diff
@@ -6,8 +6,8 @@ alias: google_tag
 
 --- filters
 ||googletagmanager.com/gtm.js^$3p
-||googletagmanager.com/a?^$3p
-||googletagmanager.com/td?^$3p
+||googletagmanager.com/a\?(.+)&gtm=^$3p
```
Comment (philipp-classen) [inline db/patterns/google_tag_manager.eno:9]: Need to test again, but I think, it needs be expressed like that (which is not a regular expression):

```suggestion
||googletagmanager.com/a?*&gtm=^$3p
```

Or alternatively, enclose it with `/`, so the adblocker will handle it as a regular expression.

File: db/patterns/google_tag_manager.eno [hunk 2]
```diff
@@ -6,8 +6,8 @@ alias: google_tag
...
+||googletagmanager.com/td?id=GTM-^$3p
```
Comment (philipp-classen) [inline db/patterns/google_tag_manager.eno:10]:
```suggestion
/^https:\/\/www\.googletagmanager\.com\/a\?(?:.+&)?gtm=/$3p
/^https:\/\/www\.googletagmanager\.com\/td\?(?:.+&)?id=GTM-/$3p
```

Notes:
* Switch to regular expression syntax
* Make no assumption about the position of the `gtm` and `id=GTM-` parameters
* Implementation detail: `?:` only to not store the capture group, since we do not access it
